web_ui/src/pages/utils.ts

```diff
@@ -134,3 +156,22 @@ export const projectPointOnLine = ([startPoint, endPoint]: ProjectLine, point: P
         y: b.y * scale + startPoint.y,
     };
 };
+
+const hasEqualLabelId = (id: string) => (point: KeypointNode) => point.label.id === id;
+
+export const getInitialKeypointStructure = ({ edges, positions }: KeypointStructure): TemplateState => {
+    const linkedEdged = edges.map(({ nodes: [fromId, toId] }) => {
+        const toLabel = positions.find(hasEqualLabelId(fromId));
+        const fromLabel = positions.find(hasEqualLabelId(toId));
+
+        return { id: uuidv4(), from: toLabel, to: fromLabel } as EdgeLine;
+    });
+
+    return { edges: linkedEdged, points: positions };
+};
+
+export const getMaxMinPoint = <T extends Point>(points: T[], pointAxis: PointAxis) => {
+    const minAxisValue = Math.min(...points.map((point) => point[pointAxis]));
+    const maxAxisValue = Math.max(...points.map((point) => point[pointAxis]));
```

Comment:
Possible suggestion: we could also use a reduce here and get the min and max in 1 loop and without copying `points`.
I guess in our case we don't expect much more than say 100 points so the performance (in other cases doing `Math.min(...xs)` can be a lot slower than doing a reduce or for loop) shouldn't matter but wanted to share in case you think it's useful.

Reply:
yes, I'll address it in my next PR